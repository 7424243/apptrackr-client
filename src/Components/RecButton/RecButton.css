/* Classes */

.recbutton {
    font-size: 15px;
    padding: 5px 12px;
    border-radius: 8px;
    border: gray;
    width: 100px;
    display: block;
    margin-top: 7px;
    color: #686d76;
    font-weight: bolder;
}

.recbutton:hover {
    box-shadow: 0 12px 16px 0 rgba(0,0,0,0.24), 0 17px 50px 0 rgba(0,0,0,0.19);
}

.recbutton_content {
    font-weight: bold;
}