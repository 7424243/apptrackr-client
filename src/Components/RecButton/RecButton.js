import React from 'react'
import {Link} from 'react-router-dom'
import './RecButton.css'

function RecButton(props) {
    return (
        <Link 
            className='recbutton_content' 
            to={props.path}
        >
            <button className='recbutton'>{props.content}</button>
        </Link>
    )
}

export default RecButton
