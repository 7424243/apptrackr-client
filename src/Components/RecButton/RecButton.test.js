import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import RecButton from './RecButton'

describe('RecButton Component', () => {

    const props = {
        path: '/', 
        content: 'test'
    }

    //smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<RecButton {...props}/>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    //snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<RecButton {...props}/>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})