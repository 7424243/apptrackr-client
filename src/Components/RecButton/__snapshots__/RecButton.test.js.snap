// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecButton Component renders the UI as expected 1`] = `
<div
  className="recbutton_content"
>
  <button
    className="recbutton"
    string="test"
  />
</div>
`;
